client/components/App.jsx

```diff
@@ -1,33 +1,124 @@
 import React from 'react';
-import TopComponent from './TopComponent.jsx'
+import TopComponent from './topComponent.jsx';
+import BottomComponent from './bottomComponent.jsx';
+import PositiveReviews from './positiveReviews.jsx';
+import NegativeReviews from './negativeReviews.jsx';
+
 
 class App extends React.Component {
   constructor(props) {
     super(props)
     this.state = {
-    	reviews: []
+    	reviews: [],
+      topFour: [],
+      botFour: []
+    }
+  }
+   
+  sortHighest(state) {
+    let a = state;
+    function compare(a,b) {
+      if (a.Upvotes < b.Upvotes)
+      return 1;
+      if (a.Upvotes > b.Upvotes)
+      return -1;
+      return 0;
+    }
+    
+    function compare2(a,b) {
+      if (a.Stars < b.Stars)
+      return 1;
+      if (a.Stars > b.Stars)
+      return -1;
+      return 0;
+    }
+
+    a.sort(compare)
+    a.sort(compare2)
+  }
+
+  sortLowest(state) {
+    let b = state;
+    function compare(a,b) {
+      if (a.Upvotes < b.Upvotes)
+      return 1;
+      if (a.Upvotes > b.Upvotes)
+      return -1;
+      return 0;
+    }
+
+    function compare2(a,b) {
+      if (a.Stars < b.Stars)
+      return -1;
+      if (a.Stars > b.Stars)
+      return 1;
+      return 0;
+    }
+
+    b.sort(compare)
+    b.sort(compare2)
+  }
+
+  getFirstFour(array) {
+    let top = []
+    for (var i = 0; i < 4; i++) {
+      top.push(array[i])
     }
+    return top
   }
 
   componentDidMount() {
   	fetch('/api/item')
```

Comment:
A suggestion: perhaps to have a bit more precision with your code, maybe you could make this fetch here a separate function and call that function within componentDidMount to make it more clear what's happening here.